scijava-types/src/main/java/org/scijava/types/DefaultTypeReifier.java

```diff
@@ -67,9 +63,13 @@ private synchronized void initExtractors() {
 		extractors = getInstances();
 	}
 
-	private List<TypeExtractor> getInstances() {
-		return discoverers.parallelStream() //
+	private Map<Class<?>, TypeExtractor> getInstances() {
+		Map<Class<?>, TypeExtractor> instances = new HashMap<>();
+		discoverers.stream() //
 			.flatMap(d -> d.discover(TypeExtractor.class).stream()) //
-			.sorted().collect(Collectors.toList());
+			// Find the correct set for each extractor and add it.
+			.sorted() //
+			.forEach(e -> instances.putIfAbsent(e.baseClass(), e));
```

Comment:
This makes it so that if there are two `TypeExtractor` plugins with the same base class, the first one *silently* wins. It would be better—if we are certain that there should never be two `TypeExtractor` plugins operating on the same base class—to issue an error or at least a warning when a second one is present.

Reply:
Good catch - I warn now in e60e711b